include/proteus/CoreLLVMDevice.hpp

```diff
@@ -111,6 +111,8 @@ inline void setKernelDims(Module &M, dim3 &GridDim, dim3 &BlockDim) {
     }
   };
 
+// These optimizations cause regressions for CUDA
+#if defined(PROTEUS_ENABLE_HIP)
```

Comment:
This causes errors for unused variable. Better make the call a no-op for CUDA:
```cpp
auto InsertAssume = ... {
// No-op for CUDA because this specialization causes performance regressions.
#if PROTEUS_ENABLE_CUDA
  return;
#endif
...
}
```